pkg/machine/machineset.go

```diff
@@ -0,0 +1,378 @@
+
+		return nil, err
+	}
+
+	return machineSet, err
+}
+
+// Exists checks whether the given MachineSet exists.
+func (builder *MachineSetBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof(
+		"Checking if MachineSet %s exists in namespace %s", builder.Definition.Name,
+		builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to collect MachineSet object due to %s", err.Error())
+	}
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Create makes a MachineSet in cluster and stores the created object in struct.
+func (builder *MachineSetBuilder) Create() (*MachineSetBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the MachineSet %s", builder.Definition.Name)
+
+	var err error
+	if !builder.Exists() {
+		builder.Object, err = builder.apiClient.MachineSets(builder.Definition.Namespace).Create(
+			context.TODO(), builder.Definition, metav1.CreateOptions{})
+	}
+
+	return builder, err
+}
+
+// Delete removes a MachineSet object from a cluster.
+func (builder *MachineSetBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the MachineSet object %s",
+		builder.Definition.Name)
+
+	if !builder.Exists() {
+		return fmt.Errorf("MachineSet cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.MachineSets(builder.Object.Namespace).Delete(
+		context.TODO(), builder.Object.Name, metav1.DeleteOptions{})
+
+	if err != nil {
+		return fmt.Errorf("cannot delete MachineSet: %w", err)
+	}
+
+	return err
+}
+
+// GetWorkerMachineSetList returns the list of worker MachineSets in a namespace on a cluster.
+func GetWorkerMachineSetList(apiClient *clients.Settings, namespace string) (*machinev1beta1.MachineSetList, error) {
+	list := &machinev1beta1.MachineSetList{
+		Items: []machinev1beta1.MachineSet{},
+	}
+
+	resp, err := apiClient.MachineSets(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, ms := range resp.Items {
+		if val, ok := ms.Spec.Template.ObjectMeta.Labels["machine.openshift.io/cluster-api-machine-role"]; ok &&
+			val == "worker" {
+			list.Items = append(list.Items, ms)
+		}
+	}
+
+	return list, nil
+}
+
+// WaitForMachineSetReady waits until MachineSet first replica is Ready.
+func WaitForMachineSetReady(apiClient *clients.Settings, namespace, machineSetName string, pollInterval,
+	timeout time.Duration) error {
+	return wait.PollImmediate(pollInterval, timeout, func() (bool, error) {
+		machineSetPulled, err := PullMachineSet(apiClient, namespace, machineSetName)
+
+		if err != nil {
+			glog.V(100).Infof("MachineSet pull from cluster error: %s\n", err)
+
+			return false, err
+		}
+
+		if machineSetPulled.Object.Status.ReadyReplicas == 1 {
```

Comment:
Why are you using 1 here? Shouldn't you just check that `ReadyReplicas == Replicas`` ?

Reply:
Done